### ISSUE TITLE
Feature Request: monitor e-mail server (SMTP, IMAP, etc)

### ISSUE THREAD

**Issue by MayeulC** [statping-ng/statping-ng#240]
Checking if a service answers on TCP port 25 is a good start, but not enough in my opinion.

It would be nice to check the response against an expected one. For instance, this is what outlook answers (from the MX server associated to outlook.com):

### SMTP relay check (port 25, usually blocked by ISPs)
```shell
$ nc hotmail-com.olc.protection.outlook.com. 25
220 DB8EUR06FT026.mail.protection.outlook.com Microsoft ESMTP MAIL Service ready at Sat, 2 Sep 2023 14:30:51 +0000
```

### IMAP check (port 143, unsecured)
```shell
$ nc imap.orange.fr 143
* OK [CAPABILITY STARTTLS IMAP4rev1 LITERAL+ SASL-IR LOGIN-REFERRALS ID ENABLE AUTH=PLAIN AUTH=LOGIN] Dovecot ready.
```

### IMAP check (port 993, over SSL/TLS -- IMAPS)
```shell
$ openssl s_client -connect imap.gmail.com:993
<snip, a lot of certificate information>
* OK Gimap ready for requests from <my ip> g5mb90881774wmo
```
note that orange.fr returns the same value via IMAPS as it does via IMAP

POP is a bit out of scope, it's barely used these days.

-----

So, this feature is a request for, in order from simplest to implement to nicest to have:
1. Check the TCP answer against a known good reference
2. Support TCP+SSL+TLS with the first point
3. Pre-configure a few more protocols
4. Maybe more in-depth checks?
